src/log.js

```diff
@@ -402,15 +401,7 @@ export class Log {
       );
     }
 
-    try {
-      const assertion = this == logs.user ? pureUserAssert : pureDevAssert;
-      return assertion.apply(null, arguments);
-    } catch (e) {
-      this.prepareError_(e);
-      // __AMP_REPORT_ERROR is installed globally per window in the entry point.
-      self.__AMP_REPORT_ERROR(e);
-      throw e;
-    }
+    return assertion.bind(null, this.suffix_).apply(null, arguments);
```

Comment:
This is actually incredibly slow to do, it'd be better to just construct an array:
```suggestion
    return assertion.apply(null, [this.suffix_].concat(Array.prototype.slice.call(arguments)));
```

Reply:
I was back and forth between these two. Good to know